src/main/java/uk/gov/hmcts/darts/common/multipart/JavaMailXmlWithFileMultiPartRequest.java

```diff
@@ -0,0 +1,191 @@
+package uk.gov.hmcts.darts.common.multipart;
+
+import jakarta.servlet.ServletInputStream;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletRequestWrapper;
+import lombok.extern.slf4j.Slf4j;
+import uk.gov.hmcts.darts.common.function.ConsumerWithIoException;
+import uk.gov.hmcts.darts.ws.CodeAndMessage;
+import uk.gov.hmcts.darts.ws.DartsException;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.List;
+import javax.mail.BodyPart;
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMultipart;
+
+
+/**
+ * A request that allows processing of incoming MTOM requests containing xml meta data and an optional binary file. This
+ * implementation streams for the most part but does write a binary file where necessary to allow retries of uploaded byte data.
+ *
+ * <p>The implementation uses java mail as a basis of MTOM parsing. It assumes one piece of xml meta data per file upload at
+ * the time of writing
+ */
+@Slf4j
+public class JavaMailXmlWithFileMultiPartRequest extends HttpServletRequestWrapper implements XmlWithFileMultiPartRequest {
+    private XmlFileUploadPart parsedData;
+
+
+    @SuppressWarnings("PMD.AvoidThrowingRawExceptionTypes")
+    public JavaMailXmlWithFileMultiPartRequest(HttpServletRequest request) {
+        super(request);
+        parse(request);
+    }
+
+    private void parse(HttpServletRequest request) {
+        try {
+            MimeMultipart mimeMultipart = new MimeMultipart(new HttpRequestDataSource(request
+            ));
+
+            BodyPart xmlPayload = MultIPartUtil.getXml(mimeMultipart);
+            if (xmlPayload == null) {
+                throw new DartsException(null, CodeAndMessage.ERROR);
+            }
+
+            BodyPart binary = MultIPartUtil.getBinary(mimeMultipart);
```

Comment:
Nope. it can be null. Its checked later on using the hadBinaryFile method. All multipart requests run through this code at the moment if we have a file or just the xml.